### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by JbirdL86**
What's new :question: 

- Add users model
- Add posts model
- Add comments model
- Add likes model
- Add current user method

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :post, foreign_key: :post_id
+  belongs_to :user, foreign_key: :author_id
```
Comment (Dipeshtwis) [inline app/models/like.rb:3]: - [x] According to the ERD, the foreign key in this like table should be AuthorID, therefore the association should be:
`belongs_to :author, class_name: 'User'`

Please consider refactoring this appropriately.

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, foreign_key: :author_id
```
Comment (Dipeshtwis) [inline app/models/post.rb:2]: - [x] According to the ERD, the foreign key in this post table should be AuthorID, therefore the association should be:
`belongs_to :author, class_name: 'User'`

Please consider refactoring this appropriately.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: :author_id
+  belongs_to :post, foreign_key: :post_id
+
+  def update_comments_count
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (Dipeshtwis) [inline app/models/comment.rb]: [OPTIONAL] 

Here, we can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_count` method every time a new post has been saved to the database, then make the `update_comments_count` method private. Your model would end up looking like this:

```ruby
# app/models/comment.rb
  after_save :update_comment_counter

  private

  def update_comment_counter
    post.increment!(:comments_counter)
  end
  ```

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: :author_id
```
Comment (Dipeshtwis) [inline app/models/comment.rb:2]: - [x] According to the ERD, the foreign key in this comment table should be AuthorID, therefore the association should be:
`belongs_to :author, class_name: 'User'`

Please consider refactoring this appropriately.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @JbirdL86, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

- [ ] Also, in the README please mention all the steps to run the project locally. Currently, we are missing two commands. `rails db:migrate` to migrate the database and `npm i or yarn i` to install all the packages. Please update it.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :+1: 

Hello, @JbirdL86   👋🏾 👋🏾 congratulations on implementing all the project requirements.

## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

**Pull request merged**